.github/actions/k-bench/evaluate/parse.py

```diff
@@ -0,0 +1,83 @@
+"""Parse logs of K-Bench tests and generate performance graphs."""
+import json
+import os
+from collections import defaultdict
+from typing import Tuple
+
+from evaluators import default, fio, network
+
+
+def configure() -> Tuple[str, str, str, str | None]:
+    """Read the benchmark data paths.
+
+    Expects ENV vars (required):
+    - KBENCH_RESULTS=/path/to/k-bench/out
+    - CSP=azure
+    - BDIR=benchmarks
+
+    Optional:
+    - EXT_NAME=AKS  # Overrides "constellation-$CSP" naming, used to parse results from managed Kubernetes
+    - GITHUB_SHA=ffac5... # Set by GitHub actions, stored in the result JSON.
+
+    Raises TypeError if at least one of them is missing.
+
+    Returns: a tuple of (base_path, csp, out_dir, ext_provider_name).
+    """
+    base_path = os.environ.get('KBENCH_RESULTS', None)
+    csp = os.environ.get('CSP', None)
+    out_dir = os.environ.get('BDIR', None)
+    if not base_path or not csp or not out_dir:
+        raise TypeError(
+            'ENV variables KBENCH_RESULTS, CSP, BDIR are required.')
+
+    ext_provider_name = os.environ.get('EXT_NAME', None)
+    commit_hash = os.environ.get('GITHUB_SHA', "N/A")
+    return base_path, csp, out_dir, ext_provider_name, commit_hash
+
+
+def main() -> None:
+    """Read and parse the K-Bench tests.
+
+    Write results of the current environment to a JSON file.
+    """
+    base_path, csp, out_dir, ext_provider_name, commit_hash = configure()
+
+    if not ext_provider_name:
+        # Constellation benchmark.
+        ext_provider_name = f"constellation-{csp}"
+
+    # Expect the results in directory:
+    # kbench-EXT_PROVIDER_NAME/
+    benchmark_path = os.path.join(
+        base_path,
+        "kbench-" + ext_provider_name,
+    )
+    tests = {ext_provider_name: benchmark_path}
+    out_file_name = "{nm}.json".format(nm=ext_provider_name)
+
+    if not os.path.exists(benchmark_path):
```

Comment:
[mkdir](https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir) can ensure the path exists if you want that. Seems more robust than failing if the path doesn't exist yet?

Reply:
Yes, cheers!

Reply:
If it's about checking if the path is correct I would expect a check using normpath/realpath with an expected value.